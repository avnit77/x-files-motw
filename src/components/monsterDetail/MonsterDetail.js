import React from 'react';
import styles from './MonsterDetail.css';
import PropTypes from 'prop-types';
import { useMonster } from '../../hooks/motw';


const MonsterDetail = ({ match }) => {
  const { selectedMonster } = useMonster(match.params.name);
  console.log(selectedMonster, 'selected Monster');

  return (
    <div className={styles.MonsterDetail}>
      <h1>{selectedMonster.name}</h1>
      <img src={selectedMonster.image}/>
      <ul>
        <li>Description: {selectedMonster.description}</li>
        <li>Status: {selectedMonster.status}</li>
        <li>Portrayed By: {selectedMonster.portrayedby}</li>
      </ul>
    </div>
  );
};

MonsterDetail.propTypes = {
  match: PropTypes.shape({
    params: PropTypes.shape({
      name: PropTypes.string.isRequired
    }).isRequired
  }).isRequired
};

export default MonsterDetail;
