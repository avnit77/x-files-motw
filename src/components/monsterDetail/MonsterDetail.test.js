import React from 'react';
import { shallow } from 'enzyme';
import MonsterDetail from './MonsterDetail';

describe('MonsterDetail component', () => {
  it('matches a snapshot', () => {
    const wrapper = shallow(<MonsterDetail match={{ params: {
      id: '1'
    } }}/>);
    expect(wrapper).toMatchSnapshot();
  });
});