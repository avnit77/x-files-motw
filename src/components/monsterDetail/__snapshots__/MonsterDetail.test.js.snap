// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MonsterDetail component matches a snapshot 1`] = `
<div
  className="MonsterDetail"
>
  <h1 />
  <img />
  <ul>
    <li>
      Description: 
    </li>
    <li>
      Status: 
    </li>
    <li>
      Portrayed By: 
    </li>
  </ul>
</div>
`;
